SocialCareCaseViewerApi/V1/Gateways/ProcessDataGateway.cs

```diff
@@ -47,10 +47,20 @@ public IEnumerable<CareCaseData> GetProcessData(ListCasesRequest request)
             }
             else
             {
-                var firstNameFilter = !string.IsNullOrWhiteSpace(request.FirstName) ?
-                    Builders<BsonDocument>.Filter.Regex("first_name", BsonRegularExpression.Create(new Regex(request.FirstName, RegexOptions.IgnoreCase))) : null;
-                var lastNameFilter = !string.IsNullOrWhiteSpace(request.LastName) ?
-                    Builders<BsonDocument>.Filter.Regex("last_name", BsonRegularExpression.Create(new Regex(request.LastName, RegexOptions.IgnoreCase))) : null;
+                if (!request.ExactNameMatch)
+                {
+                    firstNameFilter = !string.IsNullOrWhiteSpace(request.FirstName) ?
```

Comment:
we are matching names separately so it's not full name exact match but I think that's OK for now